packages/database/src/migrations/1744234388450-movePatientProgramRegistrationsToAuditTable.ts

```diff
@@ -0,0 +1,118 @@
+import { FACT_CURRENT_SYNC_TICK, SYSTEM_USER_UUID } from '@tamanu/constants';
+import { selectFacilityIds } from '@tamanu/utils/selectFacilityIds';
+import config from 'config';
+import { QueryInterface, QueryTypes } from 'sequelize';
+
+interface tableOid {
+  oid: number;
+}
+
+export async function up(query: QueryInterface): Promise<void> {
+  const [tableOidQuery]: any = await query.sequelize.query<tableOid>(
+    `SELECT oid FROM pg_class WHERE relname = 'patient_program_registrations';`,
+    { type: QueryTypes.SELECT },
+  );
+  const tableOid = tableOidQuery.oid;
+
+  // Check if updated_at_sync_tick exists in logs.changes
+  const [changesUpdatedAtSyncTickQuery]: any = await query.sequelize.query(`
+    SELECT EXISTS (SELECT TRUE
+    FROM information_schema.columns 
+    WHERE table_schema = 'logs' AND table_name = 'changes' AND column_name = 'updated_at_sync_tick');
+  `);
+  const changesHasUpdatedAtSyncTick = changesUpdatedAtSyncTickQuery?.[0]?.exists;
+  const updatedAtSyncTickSelect = `(SELECT value FROM local_system_facts WHERE key = '${FACT_CURRENT_SYNC_TICK}')::bigint,`;
+
+  // Check if updated_at_sync_tick exists in patient_program_registrations
+  const [pprUpdatedAtSyncTickQuery]: any = await query.sequelize.query(`
+    SELECT EXISTS (SELECT TRUE
+    FROM information_schema.columns 
+    WHERE table_schema = 'public' AND table_name = 'patient_program_registrations' AND column_name = 'updated_at_sync_tick');
+  `);
+  const pprHasUpdatedAtSyncTick = pprUpdatedAtSyncTickQuery?.[0]?.exists;
+  const isFacilityServer = !!selectFacilityIds(config);
+  const syncTickInitialValue = isFacilityServer ? '-999,' : '0,';
+
+  // Migrate historical changes to audit table
+  await query.sequelize.query(`
+    INSERT INTO logs.changes (
+      table_oid,
+      table_schema,
+      table_name,
+      logged_at,
+      created_at,
+      updated_at,
+      deleted_at,
+      ${changesHasUpdatedAtSyncTick ? 'updated_at_sync_tick,' : ''}
+      updated_by_user_id,
+      record_id,
+      record_update,
+      record_created_at,
+      record_updated_at,
+      record_deleted_at,
+      record_sync_tick,
+      record_data
+    )
+    SELECT
+      ${tableOid},
+      'public',
+      'patient_program_registrations',
+      to_timestamp(ppr.date, 'yyyy-mm-dd hh24:mi:ss'),
+      now(),
+      now(),
+      CASE WHEN ppr.deleted_at IS NOT NULL THEN now() ELSE NULL END,
+      ${changesHasUpdatedAtSyncTick ? updatedAtSyncTickSelect : ''}
+      COALESCE(ppr.clinician_id::text, '${SYSTEM_USER_UUID}'),
+      ppr.id,
+      registration_summary.is_insert,
+      ppr.created_at,
+      ppr.updated_at,
+      ppr.deleted_at,
+      ${pprHasUpdatedAtSyncTick ? 'ppr.updated_at_sync_tick,' : syncTickInitialValue}
+      to_jsonb(ppr.*)
```

Comment:
Should we strip this too
```suggestion
   to_jsonb(ppr.*) - 'is_most_recent',
```

Reply:
I don't see why: I wonder if there is a way to remove that is not a hassle plus migrations will happen in the future and we're not going to modify previous logs on it. It just so happens that we are also ditching that field here.